game/plugin/entity/ground-item/src/ground_item_sync.kt

```diff
@@ -0,0 +1,72 @@
+import org.apollo.game.GameConstants
+import org.apollo.game.model.entity.GroundItem
+import org.apollo.game.scheduling.ScheduledTask
+
+/**
+ * A [ScheduledTask] that manages the globalization and expiration of [GroundItem]s.
+ */
+class GroundItemSynchronizationTask(private val groundItem: GroundItem) : ScheduledTask(DELAY, false) {
```

Comment:
I think this could be better done as a single task that polls GroundItem's from a queue every tick.

Reply:
Mind explaining how that queue should work? I don't think to queue them makes much sense here as all this task is doing is keeping the ground items state in sync with the client at the appropriate time.

Reply:
So there's one GroundItemSyncTask at all times and new ones are just added to its queue, rather than creating a new task for each item. Downside of this is we have to implement what is essentially scheduling logic in the task itself

Reply:
Kind of defeats the purpose of the task scheduler, does it not?

Reply:
Partially yes; that is the downside